sklearn/utils/validation.py

```diff
@@ -632,20 +632,14 @@ def check_array(array, accept_sparse=False, *, accept_large_sparse=True,
                     "your data has a single feature or array.reshape(1, -1) "
                     "if it contains a single sample.".format(array))
 
-        # in the future np.flexible dtypes will be handled like object dtypes
-        if dtype_numeric and np.issubdtype(array.dtype, np.flexible):
-            warnings.warn(
-                "Beginning in version 0.22, arrays of bytes/strings will be "
-                "converted to decimal numbers if dtype='numeric'. "
-                "It is recommended that you convert the array to "
-                "a float dtype before using it in scikit-learn, "
-                "for example by using "
-                "your_array = your_array.astype(np.float64).",
-                FutureWarning, stacklevel=2)
-
```

Comment:
I think we should still deprecate string / bytes inputs when `dtype_numeric`, even in the case where they can be converted to numerical values with `.astype(np.float64)`. I think we should even consider doing it for object dtype arrays.

For instance in the past @jnothman said: https://github.com/scikit-learn/scikit-learn/pull/10495#pullrequestreview-91740097 and even before, in the parent issue, @amueller said: https://github.com/scikit-learn/scikit-learn/issues/10229#issuecomment-351103825

`dtype.kind == 'V'` should just error directly. Implicit conversion should probably always considered a bug as @RuRo explained in https://github.com/scikit-learn/scikit-learn/pull/18496#issuecomment-715603114.